### PR TITLE
Tech/861/split project versioning

### LINKED ISSUE #861: Split the project into visualization and analysis
# Question

I have not been with CodeCharta at the time, when people decided to put the visualization and the analysis part into one project long ago. Probably there were some good reasons, but time changes and the projects get larger day by day. Now I am just questioning if this is still the best solution or if we should consider separating the two parts into individual projects `codecharta-visualization` and `codecharta-analysis` with individual pipelines, releases etc. 

How about shortly collecting all prop's and con's in the list below? I am sure there are people who have a lot more experience than me about that topic, so feel free to edit this description/list and optionally comment your changes below, so we get a notification that something has changed 😃 

## Pro's
- Less time for travis to run the pipeline, generally means less time to wait until PR's can be merged
- Individual releases show more accurate if the desired project (visualization or analysis) changed
- Inidvidual IDE configuration to use gradle for the analysis project, but not for the visualization

## Con's
- the effort to separate the projects
- increasing the distance between both projects although they are closely related using the cc.json schema. Although this could be solved by a good README referencing the other project.

## Following questions
- Where is the CHANGELOG past gonna be? Manually separated or do both get the full CHANGELOG of the past?
- The visualisations package.json includes the `apiVersion` of the cc.json schema. Is this documented somewhere?
- Will we move the gh-pages folder into the visualization project?

### PR DESCRIPTION
**Pull request by phanlezz**
# Split Release into two: Analysis and Visualization

Closes #861

## Description

This PR updates:
- GitHub Actions
- Local Release Python Scripts
- Build Demo Scripts
- GH-Pages
... to support the separate releases in this monorepo.

## Definition of Done

A PR is only ready for merge once all the following acceptance criteria are fulfilled:
- [x] Changes have been manually tested (Python Release Script)
- [x] All TODOs related to this PR have been closed
- [x] There are automated tests for newly written code and bug fixes
- [x] All bugs discovered while working on this PR have been submitted as issues (if not already an open issue)
- [x] Documentation (GH-pages, analysis/visualization READMEs, parser READMEs, --help, etc.) has been updated (almost always necessary except for bug fixes)
- [x] CHANGELOG.md has been updated

## Todos:

- [x] Python Script
- [x] Split Changelogs
- [x] Adjust Release Workflow
	- [x] Split web demo build script
	- [x] On Push Tag update
	- [x] Npm publish block syntax
	- [x] Workflow warnings:
		- [x] working-dir -> removed not a valid input
		- [x] set-output -> updated action
	- [x] Update GH-pages in both releases -> [See comment](https://github.com/MaibornWolff/codecharta/pull/3499#issuecomment-1956429280)
	- [x] Docker Release Version
- [x] cc.json API version in visualization -> already in root
- [x] GH-pages release blog separated for analysis and visualization -> will stay in current format (equal to GitHub)
- [x] Visualization new release popup
- [x] Check test.yml
- [x] Actions not running if based on fork -> created issue #3500
- [x] ~~Separate steps in GH-action release yml~~

### CODE REVIEW

**Comment by phanlezz** [comment]
Preview release entry:
grafik

**Review by Nereboss — APPROVED** [review]
LGTM 🚀

File: script/make_release.py [hunk 1]
```diff
@@ -47,13 +49,53 @@ def get_latest_changelog_entries(path):
...
+            fp.write(
+                f"## [{new_version}] - {date_formatted}\n")
+        else:
+            fp.write(line)
+
+  with in_place.InPlace(changelog_path, encoding="utf-8") as fp:
+    line_number = 0
+
+    for line in fp:
+        if line_number == 6:
```
Comment (ce-bo) [inline script/make_release.py:72]: This seems to be ok, because it is unlikely that anybody adds something before line 6.

Comment (phanlezz) [reply script/make_release.py:72]: I didn't change this. It is as before.

File: script/make_release.py [hunk 2]
```diff
@@ -47,13 +49,53 @@ def get_latest_changelog_entries(path):
...
+            fp.write(line)
+        line_number = line_number + 1
+
+def update_readme(readme_path):
+  with in_place.InPlace(readme_path, encoding="utf-8") as fp:
+    line_number = 0
+
+    # line_number in array format -> first line = line 0
+    for line in fp:
+      if line_number == 24:
```
[the remark this answers is not in the document]
Comment (phanlezz) [reply script/make_release.py:85]: There isn't that much content before that in the readme. 
It's the best way to point to the latest releases for both subrepos.

[the remark this answers is not in the document]
Comment (phanlezz) [reply script/make_release.py:85]: I will remove it, if you want to.

[the remark this answers is not in the document]
Comment (phanlezz) [reply script/make_release.py:85]: Some workarounds:

- I could add a large comment block, warning devs not change the line
- We could add a rule to husky, warning devs if changes have been made to the readme to check the line
- Move it to line 1

[the remark this answers is not in the document]
Comment (phanlezz) [reply script/make_release.py:85]: What do you think @ce-bo

**Pull request merged**